web/source/dom/domEventHandlers.ts

```diff
@@ -734,6 +675,156 @@ namespace com.keyman.dom {
+          // If this block executed, cpMax will be 'tight' when complete.
+        }
+
+        // The caret itself should now lie within the target row.  The start of the row must be at or before
+        // the caret, while the end of the row must be at or after it.
+        //
+        // To properly reuse the InputHTMLElement-based x-coord search, we need the bounds to be TIGHT after
+        // this comment's containing block finishes. As in, we need [cpMin, cpMax] === [start of row, end of row].
+        //
+        // Note:  in certain scenarios, one of the bounds can be VERY far away at this point.
+        // Say, if a new touch event wants to move the caret nearby the previous location, as the
+        // bound on the opposite side may not have gotten overwritten even once so far in this call!
+        // Therefore... more binary search.
+
+        // Determine minimum caret position on the target row.
+        let minCpMin = cpMin; // Is before the row.
+        let maxCpMin = cp;
+        while(maxCpMin - minCpMin > 1) {
+          const searchPt = Math.round((maxCpMin+minCpMin)/2);
+          target.setTextCaret(searchPt);
+
+          const y=dom.Utils.getAbsoluteY(caret)-dy;  //top of caret
+          if(y < minY) {
+            minCpMin = searchPt+1; // We already know it's not on this row, so the minimum possible should be increased.
+          } else {
+            maxCpMin = searchPt; // Still in the same row, so the boundary can only be at or before this point.
+          }
+          cpMin = searchPt;
+        }
+
+        // Determine maximum caret position on the target row.
+        let minCpMax = cp;
+        let maxCpMax = cpMax; // Is after the row.
+        while(maxCpMax - minCpMax > 1) {
+          const searchPt = Math.round((maxCpMax+minCpMax)/2);
+          target.setTextCaret(searchPt);
+
+          const y=dom.Utils.getAbsoluteY(caret)-dy;  //top of caret
+          if(y > maxY) {
+            maxCpMax = searchPt-1; // We already know it's not on this row, so the maximum possible should be decreased.
+          } else {
+            minCpMax = searchPt; // Still in the same row, so the boundary can only be at or after this point.
+          }
+          cpMax = searchPt;
+        }
+
+        // Set the potential caret in the middle of the range.
+        cp = Math.round((cpMin + cpMax)/2);
+        target.setTextCaret(cp);
+      }
+
+      // Caret repositioning for horizontal scrolling of RTL text
+
+      // snapOrder - 'snaps' the touch location in a manner corresponding to the 'ltr' vs 'rtl' orientation.
+      // Think of it as performing a floor() function, but the floor depends on the origin's direction.
+      var snapOrder;
+      if(target.dir == 'rtl') {  // I would use arrow functions, but IE doesn't like 'em.
+        snapOrder = function(a, b) {
+          return a < b;
+        };
+      } else {
+        snapOrder = function(a, b) {
+          return a > b;
+        };
+      }
```

Comment:
```suggestion
      const snapOrder = target.dir == 'rtl' ? (a,b) => a < b : (a,b) => a > b;
```

Reply:
Would it be fine if I placed this change within #6557 instead?

Reply:
Sure